### PR TITLE
Improve Font Color assignment in tree map chart component

### PR DESCRIPTION
**Pull request by VeerChaurasia**
Improve Font Color and background color

### CODE REVIEW

File: libs/ui/src/lib/treemap-chart/treemap-chart.component.ts [hunk 1]
```diff
@@ -35,6 +35,11 @@ import { NgxSkeletonLoaderModule } from 'ngx-skeleton-loader';
 
 const { gray, green, red } = require('open-color');
 
+interface ColorConfig {
```
Comment (dtslvr) [inline libs/ui/src/lib/treemap-chart/treemap-chart.component.ts:38]: Please move to `libs/ui/src/lib/treemap-chart/interfaces/interfaces.ts`

File: libs/ui/src/lib/treemap-chart/treemap-chart.component.ts [hunk 2]
```diff
@@ -64,20 +69,67 @@ export class GfTreemapChartComponent
     Chart.register(LinearScale, Tooltip, TreemapController, TreemapElement);
   }
 
-  public ngAfterViewInit() {
```
Comment (dtslvr) [inline libs/ui/src/lib/treemap-chart/treemap-chart.component.ts:67]: Why has this been removed?

File: libs/ui/src/lib/treemap-chart/treemap-chart.component.ts [hunk 3]
```diff
@@ -64,20 +69,67 @@ export class GfTreemapChartComponent
...
+  ): ColorConfig {
+    if (Math.abs(annualizedNetPerformancePercent) === 0) {
+      return {
+        backgroundColor: gray[3],
+        fontColor: gray[9]
+      };
     }
-  }
 
-  public ngOnChanges() {
```
Comment (dtslvr) [inline libs/ui/src/lib/treemap-chart/treemap-chart.component.ts:73]: Why has this been removed?

File: libs/ui/src/lib/treemap-chart/treemap-chart.component.ts [hunk 4]
```diff
@@ -64,20 +69,67 @@ export class GfTreemapChartComponent
...
+      } else if (
+        annualizedNetPerformancePercent <=
+        negativeRange.min + range * 0.75
+      ) {
+        backgroundIndex = 5;
+      } else {
+        backgroundIndex = 3;
+      }
 
-  public ngOnDestroy() {
```
Comment (dtslvr) [inline libs/ui/src/lib/treemap-chart/treemap-chart.component.ts:79]: Why has this been removed?

File: libs/ui/src/lib/treemap-chart/treemap-chart.component.ts [hunk 5]
```diff
@@ -147,66 +199,47 @@ export class GfTreemapChartComponent
...
+                  )
+                }).toNumber();
+
+              // Round to 2 decimal places
+              annualizedNetPerformancePercentWithCurrencyEffect =
+                Math.round(
+                  annualizedNetPerformancePercentWithCurrencyEffect * 100
+                ) / 100;
+
+              const colorConfig = this.getColorConfig(
```
Comment (dtslvr) [inline libs/ui/src/lib/treemap-chart/treemap-chart.component.ts:235]: `const { fontColor } = this.getColorConfig(`

File: libs/ui/src/lib/treemap-chart/treemap-chart.component.ts [hunk 6]
```diff
@@ -147,66 +199,47 @@ export class GfTreemapChartComponent
...
-                negativeNetPerformancePercentsRange.min + range * 0.5
-              ) {
-                return red[7];
-              } else if (
-                annualizedNetPerformancePercentWithCurrencyEffect <=
-                negativeNetPerformancePercentsRange.min + range * 0.75
-              ) {
-                return red[5];
-              }
+            const colorConfig = this.getColorConfig(
```
Comment (dtslvr) [inline libs/ui/src/lib/treemap-chart/treemap-chart.component.ts:202]: `const { backgroundColor } = this.getColorConfig(`

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thank you for the contribution, @VeerChaurasia.
I have reviewed the changes and added a few comments.

**Comment by VeerChaurasia** [comment]
Hey @dtslvr can you pls review it

**Review by dtslvr — APPROVED** [review]
Thanks for your contribution, @VeerChaurasia! 👍🏻
I have slightly adapted the code. Ready to merge now.

**Pull request merged**